### PR TITLE
LPS-130473 Audit and refactor tests for component Tags tests to use SACRED model | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-130473

cc @david-gutierrez-mesa 
cc @Tim-Cao 
Rebased from #7293

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase [hunk 1]
```diff
@@ -29,260 +29,333 @@ definition {
 		}
 	}
 
+	@description = "The user cannot add a tag with an empty name."
 	@priority = "4"
-	test CannotAddEmptyTag {
-		Tag.addWithNullTitleCP(tagName = " ");
+	@uitest
+	test CannotAddWithEmptyName {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase:35]: CannotAddTagWithoutName

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase [hunk 2]
```diff
@@ -29,260 +29,333 @@ definition {
...
-		JSONAssettag.addTag(
-			groupName = "Guest",
-			tagName = "tagname1");
-
-		Tag.openTagsAdmin(siteURLKey = "Guest");
-
-		Tag.viewCP(tagName = "tagname1");
+	@description = "The user can delete a single tag via menu item."
+	@priority = "5"
+	test DeleteTagViaMenuItem {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase:125]: DeleteTagViaMenu

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase [hunk 3]
```diff
@@ -29,260 +29,333 @@ definition {
...
+				value1 = "20 Entries Per page");
 
-			var index = ${index} + 1;
+			Pagination.viewResults(results = "Showing 1 to 20 of 24 entries.");
 		}
 
-		ProductMenu.gotoPortlet(
-			category = "Categorization",
-			portlet = "Tags");
+		task ("View pagination results for 4 entries") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase:342]: Change the amount of items per page to 4

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase [hunk 4]
```diff
@@ -29,260 +29,333 @@ definition {
...
+			Pagination.changePagination(itemsPerPage = "4");
 
-		AssertTextEquals.assertPartialText(
-			locator1 = "Pagination#ITEMS_PER_PAGE_SELECT",
-			value1 = "20 Entries Per page");
+			Pagination.viewResults(results = "Showing 1 to 4 of 24 entries.");
+		}
+
+		task ("View pagination results for 8 entries") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase:348]: Change the amount of items per page to 8

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase [hunk 5]
```diff
@@ -29,260 +29,333 @@ definition {
...
+			Pagination.changePagination(itemsPerPage = "8");
 
-		Pagination.viewResults(results = "Showing 1 to 20 of 24 entries.");
+			Pagination.viewResults(results = "Showing 1 to 8 of 24 entries.");
+		}
 
-		Pagination.changePagination(itemsPerPage = "4");
+		task ("View pagination results for 40 entries") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/Tags.testcase:354]: Change the amount of items per page to 40

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 6]
```diff
@@ -35,612 +35,506 @@ definition {
...
+	test DisplayTagNumberTemplate {
+		task ("Add a Tag Filter to page") {
+			JSONLayout.addWidgetToPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Test Page Name",
+				widgetName = "Tag Filter");
+		}
 
-		TagsNavigationPortlet.configureDisplayStylePG(displayTemplate = "Number");
+		task ("Configure Number as display template") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:228]: Configure the display template to Number

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 7]
```diff
@@ -35,612 +35,506 @@ definition {
...
+				dmDocumentTitle = "DM Document Title",
+				groupName = "Test Site Name",
+				mimeType = "text/plain",
+				sourceFileName = "Document_1.txt");
+		}
 
-		TagsNavigationPortlet.clickTagView(
-			tagAssetCount = "first tag (2)",
-			tagName = "first tag");
+		task ("View number of usages on tags") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:268]: View the usages of tags

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 8]
```diff
@@ -35,612 +35,506 @@ definition {
...
-		TagsNavigationPortlet.clickTagView(
-			tagAssetCount = "second tag (1)",
-			tagName = "second tag");
+			TagsNavigationPortlet.clickTagView(
+				tagAssetCount = "tagname2 (1)",
+				tagName = "tagname2");
+		}
 	}
 
+	@description = "View unused tags in Tag Filter widget."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:283]: The Tag Filter widget could display unused tags

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 9]
```diff
@@ -35,612 +35,506 @@ definition {
...
-				tagName = "${tagName}");
+		task ("Add a Tag Filter to page") {
+			JSONLayout.addWidgetToPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Test Page Name",
+				widgetName = "Tag Filter");
 		}
 
-		Navigator.gotoPage(pageName = "Tags Navigation Page");
+		task ("Check show unused tags in Tag Filter configuration") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:301]: Configure the Tag Filter to display unused tags

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 10]
```diff
@@ -35,612 +35,506 @@ definition {
...
-			groupName = "Site Name",
-			layoutName = "Page Name",
-			widgetName = "Blogs");
-
-		HeadlessBlog.addBlogEntry(
-			assetTagNames = "tag name 1,tag name 2,tag name 3",
-			content = "Entry Content",
-			groupName = "Site Name",
-			title = "Entry Title");
+	test SearchMultipleTagsViaBlogs {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:318]: SearchTagsViaBlogs

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 11]
```diff
@@ -35,612 +35,506 @@ definition {
...
 	}
 
-	@priority = "5"
-	test RemoveTagViaBlogs {
+	@description = "View selected and removed tags in Blogs widget."
+	@priority = "3"
+	test ViewRemovedTagViaBlogs {
 		property portal.acceptance = "true";
-		property testray.component.names = "Blogs,Tags";
+		property testray.component.names = "Blogs";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:373]: Keep the original, this one will override the global property

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 12]
```diff
@@ -35,612 +35,506 @@ definition {
...
+				var i = ${i} + 1;
+			}
+		}
 	}
 
-	@priority = "5"
-	test RemoveTagViaBlogs {
+	@description = "View selected and removed tags in Blogs widget."
+	@priority = "3"
+	test ViewRemovedTagViaBlogs {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:371]: RemoveSelectedTagViaBlogs

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase [hunk 13]
```diff
@@ -35,612 +35,506 @@ definition {
...
 
-		AssertTextNotPresent(value1 = "tag name 3");
+				var i = ${i} + 1;
+			}
+		}
 	}
 
-	@priority = "5"
-	test RemoveTagViaBlogs {
+	@description = "View selected and removed tags in Blogs widget."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/tags/TagsUseCase.testcase:369]: The user could remove selected tag via Blogs

**Review by Tim-Cao — COMMENTED** [review]
Hi @ozysouza, would you mind modifing according to comments? Thanks

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind modifing according to comments? Thanks

Hi @Tim-Cao! I've pushed some extra commits with the changes you requested. Would you have a look at? Thanks!

**Comment by Tim-Cao** [comment]
LGTM!

**Comment by Tim-Cao** [comment]
ci:forward

**Pull request closed without merging**